x-pack/plugins/cloud_security_posture/public/pages/findings/findings.tsx

```diff
@@ -20,16 +21,62 @@ import { Redirect, useHistory, useLocation, matchPath } from 'react-router-dom';
 import { Routes, Route } from '@kbn/shared-ux-router';
 import { Configurations } from '../configurations';
 import { cloudPosturePages, findingsNavigation } from '../../common/navigation/constants';
+import { LOCAL_STORAGE_FINDINGS_LAST_SELECTED_TAB_KEY } from '../../common/constants';
+import { VULNERABILITIES_INDEX_NAME, FINDINGS_INDEX_NAME } from '../../../common/constants';
+import { useCspSetupStatusApi } from '../../common/api/use_setup_status_api';
+import { getStatusForIndexName } from '../../../common/utils/helpers';
 import { Vulnerabilities } from '../vulnerabilities';
 
+type FindingsTabKey = 'vuln_mgmt' | 'configurations';
+
+const FindingsTabRedirecter = ({ lastTabSelected }: { lastTabSelected?: FindingsTabKey }) => {
+  const location = useLocation();
+  const getSetupStatus = useCspSetupStatusApi();
+
+  if (!getSetupStatus.data) {
+    return null;
+  }
+
+  const vulnStatus = getStatusForIndexName(VULNERABILITIES_INDEX_NAME, getSetupStatus.data);
+  const findingsStatus = getStatusForIndexName(FINDINGS_INDEX_NAME, getSetupStatus.data);
+  const hasVulnerabilities = vulnStatus === 'not-empty';
+  const hasFindings = findingsStatus === 'not-empty';
+
+  // if the user has not yet made a tab selection
+  // switch to misconfigurations page if there are misconfigurations, and no vulnerabilities
+  const redirectToMisconfigurationsTab =
+    lastTabSelected === 'configurations' ||
+    (!lastTabSelected && !hasVulnerabilities && hasFindings);
+
+  if (redirectToMisconfigurationsTab) {
+    return (
+      <Redirect
+        to={{ search: location.search, pathname: findingsNavigation.findings_default.path }}
+      />
+    );
+  }
+
+  // otherwise stay on the vulnerabilities tab, since it's the first one.
+  return (
+    <Redirect to={{ search: location.search, pathname: findingsNavigation.vulnerabilities.path }} />
+  );
+};
+
 export const Findings = () => {
   const history = useHistory();
   const location = useLocation();
 
+  // restore the users most recent tab selection
+  const [lastTabSelected, setLastTabSelected] = useLocalStorage<FindingsTabKey>(
+    LOCAL_STORAGE_FINDINGS_LAST_SELECTED_TAB_KEY
+  );
+
   const navigateToVulnerabilitiesTab = () => {
+    setLastTabSelected('vuln_mgmt');
```

Comment:
I tested it locally and it looks good

https://github.com/elastic/kibana/assets/19270322/da924d67-2294-4650-8811-758bb5f9c49b

Just think we could go up a step further and call the `setLastTabSelected` on the Tab page component when it mounts instead, so it always remembers the last page the user was independently from the tab clicking, and that covers scenarios where users interact with multiple integrations (Vulnerabilities and Misconfigurations), WDYT?

https://github.com/elastic/kibana/assets/19270322/037884e5-ad49-4a8c-8544-0c1ff548895e